[
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1728112607.754579",
        "client_msg_id": "389326c6-a860-4935-9ea7-068c80daaf84",
        "text": "I am going to modify the patch to just restart the stream after 5 minutes on a heartbeat, ie no data coming through the stream, it used to check the cache (`len(self._caches) == 0`) but obviously something else is going on, I haven't heard from <@UNW8Q88EL>\n\n```def _on_heartbeat(self, data: dict) -&gt; None:\n    self._update_clk(data)\n\n    # reset stream if no data and started over 5 minutes ago\n    seconds_since_start = (\n        datetime.datetime.utcnow() - self.time_created\n    ).total_seconds()\n    if seconds_since_start &gt; 300:\n        markets = len(self._caches)\n        logger.warning(\n            \"RaceStream no data, restarting\",\n            extra={\"markets\": markets, \"seconds_since_start\": seconds_since_start},\n        )\n        raise exceptions.ListenerError(\"\", \"\")```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1727985786.582959",
        "parent_user_id": "UUE6E1LA1",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "EoFr9",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "I am going to modify the patch to just restart the stream after 5 minutes on a heartbeat, ie no data coming through the stream, it used to check the cache ("
                            },
                            {
                                "type": "text",
                                "text": "len(self._caches) == 0",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": ") but obviously something else is going on, I haven't heard from "
                            },
                            {
                                "type": "user",
                                "user_id": "UNW8Q88EL"
                            },
                            {
                                "type": "text",
                                "text": "\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "def _on_heartbeat(self, data: dict) -> None:\n    self._update_clk(data)\n\n    # reset stream if no data and started over 5 minutes ago\n    seconds_since_start = (\n        datetime.datetime.utcnow() - self.time_created\n    ).total_seconds()\n    if seconds_since_start > 300:\n        markets = len(self._caches)\n        logger.warning(\n            \"RaceStream no data, restarting\",\n            extra={\"markets\": markets, \"seconds_since_start\": seconds_since_start},\n        )\n        raise exceptions.ListenerError(\"\", \"\")"
                            }
                        ],
                        "border": 0
                    }
                ]
            }
        ],
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "U05V9E7JA58"
                ],
                "count": 1
            }
        ]
    },
    {
        "user": "UGV299K6H",
        "type": "message",
        "ts": "1728112956.860649",
        "client_msg_id": "783001c3-b7c7-403d-b90e-b10cc55b0629",
        "text": "Yeah that's obviously a better solution. Thanks <@U4H19D1D2>.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g3a262a3cfba",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/b3a262a3cfba292c18acf5777f65908a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-72.png",
            "first_name": "Michael",
            "real_name": "Michael C",
            "display_name": "Michael",
            "team": "T4G9NBD2M",
            "name": "michael",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1727985786.582959",
        "parent_user_id": "UUE6E1LA1",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "P6nXM",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Yeah that's obviously a better solution. Thanks "
                            },
                            {
                                "type": "user",
                                "user_id": "U4H19D1D2"
                            },
                            {
                                "type": "text",
                                "text": "."
                            }
                        ]
                    }
                ]
            }
        ],
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "U05V9E7JA58"
                ],
                "count": 1
            }
        ]
    },
    {
        "user": "UUE6E1LA1",
        "type": "message",
        "ts": "1728125984.081729",
        "client_msg_id": "6b468c1c-2b53-472c-92f4-afc64eaad646",
        "text": "I've not checked every race but I've not seen any status probs according to the TPD status page so assume this is a 100% on the Betfair end.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gf53cae45edd",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/f53cae45edd7f7b67bbe98819597612e.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0017-72.png",
            "first_name": "",
            "real_name": "D C",
            "display_name": "D C",
            "team": "T4G9NBD2M",
            "name": "oddsvantage",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1727985786.582959",
        "parent_user_id": "UUE6E1LA1",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "tPBlC",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "I've not checked every race but I've not seen any status probs according to the TPD status page so assume this is a 100% on the Betfair end."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1728127221.412879",
        "client_msg_id": "94FA354C-FE1A-4795-A5E6-058E59839D75",
        "text": "I think that is the only thing we can be 100% sure on ",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1727985786.582959",
        "parent_user_id": "UUE6E1LA1",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "wqrzg",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "I think that is the only thing we can be 100% sure on "
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UUE6E1LA1",
        "type": "message",
        "ts": "1728127330.286729",
        "client_msg_id": "e003607f-9d4e-4c59-9041-69a03f43831b",
        "text": "Did some\/most of us experience this at the same times? I was first hit on my recording logger stream at about 17:11 UTC on the 3rd (at least that is the timestamp of the last file alteration on that server). I'm pretty sure this was the same time on my active bot too. I messaged <@UUCD6P13J> at the time but he was unaffected until yesterday - and I was unaffected yesterday.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gf53cae45edd",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/f53cae45edd7f7b67bbe98819597612e.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0017-72.png",
            "first_name": "",
            "real_name": "D C",
            "display_name": "D C",
            "team": "T4G9NBD2M",
            "name": "oddsvantage",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1727985786.582959",
        "parent_user_id": "UUE6E1LA1",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "\/39ow",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Did some\/most of us experience this at the same times? I was first hit on my recording logger stream at about 17:11 UTC on the 3rd (at least that is the timestamp of the last file alteration on that server). I'm pretty sure this was the same time on my active bot too. I messaged "
                            },
                            {
                                "type": "user",
                                "user_id": "UUCD6P13J"
                            },
                            {
                                "type": "text",
                                "text": " at the time but he was unaffected until yesterday - and I was unaffected yesterday."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UUCD6P13J",
        "type": "message",
        "ts": "1728127514.915189",
        "client_msg_id": "7c09a646-87ce-484b-8b89-20809c5dd52e",
        "text": "Looks the same time i was affected but it was only on one instance so I didn’t notice. Think I need to improve my monitoring.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g6898272ab81",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/6898272ab810e419c2c9ca9af1e53da8.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0012-72.png",
            "first_name": "Lee",
            "real_name": "Lee U",
            "display_name": "Lee",
            "team": "T4G9NBD2M",
            "name": "lunswor",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1727985786.582959",
        "parent_user_id": "UUE6E1LA1",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "gRLoh",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Looks the same time i was affected but it was only on one instance so I didn’t notice. Think I need to improve my monitoring."
                            }
                        ]
                    }
                ]
            }
        ]
    }
]